query.c

```diff
@@ -1621,6 +1661,14 @@ query_add_optional(query_type *q, nsd_type *nsd)
 		if (q->edns.dnssec_ok)	edns->ok[7] = 0x80;
 		else			edns->ok[7] = 0x00;
 		buffer_write(q->packet, edns->ok, OPT_LEN);
+
+		/* Add Extended DNS Error (RFC8914)
+		 * to verify that we stay in bounds */
+		if (q->edns.ede >= 0)
+			q->edns.opt_reserved_space +=
+				6 + ( q->edns.ede_text_len
+			            ? q->edns.ede_text_len - 1 : 0);
```

Comment:
And also at 1692 and 1697 and 1699, why the -1?  Do you count the 0 terminator in the text_len? If so, write what the value really contains in the comment with the definition of text_len in the header file.  Or do not count that, but then the `-1` and `>1` should not be there.

Reply:
Yes sizeof(<string literal>) does include the 0 character yes

Reply:
resolved in commit fcafc1a